homeassistant/components/update/__init__.py

```diff
@@ -0,0 +1,355 @@
+    __in_progress: bool = False
+
+    @property
+    def current_version(self) -> str | None:
+        """Version currently in use."""
+        return self._attr_current_version
+
+    @property
+    def device_class(self) -> UpdateDeviceClass | str | None:
+        """Return the class of this entity."""
+        if hasattr(self, "_attr_device_class"):
+            return self._attr_device_class
+        if hasattr(self, "entity_description"):
+            return self.entity_description.device_class
+        return None
+
+    @property
+    def entity_category(self) -> EntityCategory | str | None:
+        """Return the category of the entity, if any."""
+        if hasattr(self, "_attr_entity_category"):
+            return self._attr_entity_category
+        if hasattr(self, "entity_description"):
+            return self.entity_description.entity_category
+        return EntityCategory.CONFIG
+
+    @property
+    def in_progress(self) -> bool | int | None:
+        """Update installation progress.
+
+        Needs SUPPORT_PROGRESS flag to be set for it to be used.
+
+        Can either return a boolean (True if in progress, False if not)
+        or an integer to indicate the progress in from 0 to 100%.
+        """
+        return self._attr_in_progress
+
+    @property
+    def latest_version(self) -> str | None:
+        """Latest version available for install."""
+        return self._attr_latest_version
+
+    @property
+    def release_summary(self) -> str | None:
+        """Summary of the release notes or changelog.
+
+        This is not suitable for long changelogs, but merely suitable
+        for a short excerpt update description of max 255 characters.
+        """
+        return self._attr_release_summary
+
+    @property
+    def release_url(self) -> str | None:
+        """URL to the full release notes of the latest version available."""
+        return self._attr_release_url
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return self._attr_supported_features
+
+    @property
+    def title(self) -> str | None:
+        """Title of the software.
+
+        This helps to differentiate between the device or entity name
+        versus the title of the software installed.
+        """
+        return self._attr_title
+
+    @final
+    async def async_skip(self) -> None:
+        """Skip the current offered version to update."""
+        if (latest_version := self.latest_version) is None:
+            raise HomeAssistantError(f"Cannot skip an unknown version for {self.name}")
+        if self.current_version == latest_version:
+            raise HomeAssistantError(f"No update available to skip for {self.name}")
+        self.__skipped_version = latest_version
+        self.async_write_ha_state()
+
+    async def async_install(
+        self,
+        version: str | None = None,
+        backup: bool | None = None,
+    ) -> None:
```

Comment:
While looking at the dev docs, I noticed this one.
I wonder if we should use kwargs for the install methods to make it more flexible with changes in the future and less prone to breaking.

```python
    async def async_install(
        self,
        *,
        version: str | None = None,
        backup: bool | None = None,
        **kwargs,
    ) -> None:
```
That way the integrations that does not support version/backup can just do 
```python
async def async_install(self, **kwargs) -> None:
```

Reply:
> I wonder if we should use kwargs for the install methods to make it more flexible with changes in the future and less prone to breaking.

Yeah makes sense, will adjust 👍 

> That way the integrations that does not support version/backup can just do

Typewise they can't, as it has to match the inherited signature.

Reply:
Ah... right :/